lib/new_relic/cli/commands/deployments.rb

```diff
@@ -109,22 +109,76 @@ def run
     end
   end
 
+  def api_v1?
+    @api_key.nil? || @api_key.empty?
+  end
+
   private
 
+  def create_request(uri, headers, content_type)
+    Net::HTTP::Post.new(uri, headers).tap do |req|
+      req.content_type = content_type
+    end
+  end
+
+  def application_id
+    return @application_id if @application_id
+
+    # Need to connect to collector to acquire application id from the connect response
+    # but set monitor_mode false because we don't want to actually report anything
+    begin
+      NewRelic::Agent.manual_start(monitor_mode: false)
+      NewRelic::Agent.agent.connect_to_server
+      @application_id = NewRelic::Agent.config[:primary_application_id]
+    ensure
+      NewRelic::Agent.shutdown
+    end
+  end
+
+  def set_params_v1(request)
+    params = {}
+    {
+      :application_id => @appname,
+      :host => NewRelic::Agent::Hostname.get,
+      :description => @description,
+      :user => @user,
+      :revision => @revision,
+      :changelog => @changelog
+    }.each do |k, v|
+      params["deployment[#{k}]"] = v unless v.nil? || v == ''
+    end
+    request.set_form_data(params)
+  end
+
+  def set_params_v2(request)
+    params = {
+      "deployment" => {
+        :description => @description,
+        :user => @user,
+        :revision => @revision,
+        :changelog => @changelog
+      }
+    }
+    request.body = JSON.dump(params)
```

Comment:
We could completely avoid the need for creating a `params` variable here given that we don't do anything with the hash contents after instantiation. And in doing so, the `JSON.dump` approach starts to look a bit more odd so I say we replace that with a `#to_json` call. Combining both of these we'd get this:

```diff
- params = {
+ request.body = {
    "deployment" => {
      :description => @description,
      :user => @user,
      :revision => @revision,
      :changelog => @changelog
    }
- }
+ }.to_json
- request.body = JSON.dump(params)
```

Reply:
addressed 5b7e76cc0f08701e2b5da74305327b312c74c59b